File: src/mvmctl/models/key.py
"""SSH key data models."""

from __future__ import annotations

from dataclasses import dataclass

from mvmctl.utils.common import CommonUtils


@dataclass
class SSHKeyItem:
    """SSH key record — maps to ssh_keys table."""

    id: str
    name: str
    fingerprint: str
    algorithm: str
    comment: str
    public_key_path: str
    is_default: bool
    is_present: bool
    created_at: str
    updated_at: str

    private_key_path: str | None = None

    def __post_init__(self) -> None:
        """Coerce bool fields loaded from SQLite."""
        CommonUtils.coerce_bool_fields(self, {"is_default", "is_present"})

File: pkg/api/responses/key.go
package responses

// KeyInfo groups key metadata in an inspect response.
type KeyInfo struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Fingerprint string `json:"fingerprint"`
	Algorithm   string `json:"algorithm"`
	Comment     string `json:"comment"`
	IsDefault   bool   `json:"is_default"`
	IsPresent   bool   `json:"is_present"`
}

// KeyFilesInfo groups key file paths in an inspect response.
type KeyFilesInfo struct {
	PublicKeyPath  string  `json:"public_key_path"`
	PrivateKeyPath *string `json:"private_key_path"`
}

// KeyTimestampsInfo groups key timestamps in an inspect response.
type KeyTimestampsInfo struct {
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

// KeyInspect is the structured response for key inspection.
type KeyInspect struct {
	Key        KeyInfo           `json:"key"`
	Files      KeyFilesInfo      `json:"files"`
	Timestamps KeyTimestampsInfo `json:"timestamps"`
}
